vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/SupportedCredentialFormat.kt

```diff
@@ -107,19 +107,90 @@ data class SupportedCredentialFormat(
     @SerialName("display")
     val display: Set<DisplayProperties>? = null,
 ) {
-    var isoClaims: Map<String, Map<String, RequestedCredentialClaimSpecification>>?
+    val isoClaims: Map<String, Map<String, RequestedCredentialClaimSpecification>>?
         get() = claims?.let {
-            jsonSerializer.decodeFromJsonElement<Map<String, Map<String, RequestedCredentialClaimSpecification>>>(it)
-        }
-        set(value) {
-            claims = jsonSerializer.encodeToJsonElement(value)
+            runCatching {
+                jsonSerializer.decodeFromJsonElement<Map<String, Map<String, RequestedCredentialClaimSpecification>>>(it)
+            }.getOrNull()
         }
 
-    var sdJwtClaims: Map<String, RequestedCredentialClaimSpecification>?
+    val sdJwtClaims: Map<String, RequestedCredentialClaimSpecification>?
         get() = claims?.let {
-            jsonSerializer.decodeFromJsonElement<Map<String, RequestedCredentialClaimSpecification>>(it)
-        }
-        set(value) {
-            claims = jsonSerializer.encodeToJsonElement(value)
+            runCatching {
+                jsonSerializer.decodeFromJsonElement<Map<String, RequestedCredentialClaimSpecification>>(it)
+            }.getOrNull()
         }
+
+    companion object {
```

Comment:
if all these are mutually exclusive (which I understand they are), then this is the way to go!